server/datastore/sql/index.ts

```diff
@@ -59,22 +63,52 @@ export class SqlDataStore implements Datastore {
     );
   }
 
-  getPost(id: string): Promise<Post | undefined> {
-    throw new Error('Method not implemented.');
+  async getPost(id: string): Promise<Post | undefined> {
+    return await this.db.get<Post>('SELECT * FROM posts WHERE postId = ?', id);
+  }
+
+  async deletePost(id: string): Promise<void> {
+    await this.db.run('Delete FROM posts WHERE id=?', id);
   }
-  deletePost(id: string): Promise<void> {
-    throw new Error('Method not implemented.');
+
+  async createLike(like: Like): Promise<void> {
+    await this.db.run('INSERT INTO Likes(userId,postId) VALUES(?,?)', like.userId, like.postId);
+  }
+
+  async createComment(comment: Comment): Promise<void> {
+    await this.db.run(
+      'INSERT INTO Comments(userId,postId,comment,postedAt) VALUES(?,?,?,?)',
+      comment.userId,
+      comment.postId,
+      comment.comment,
+      comment.postedAt
+    );
   }
-  createLike(like: Like): Promise<void> {
-    throw new Error('Method not implemented.');
+
+  async listComments(postId: string): Promise<Comment[]> {
+    return await this.db.all<Comment[]>('SELECT * FROM comments WHERE postId=?', postId);
   }
-  createComment(comment: Comment): Promise<void> {
-    throw new Error('Method not implemented.');
+
+  async deleteComment(id: string): Promise<void> {
+    await this.db.run('Delete FROM comments WHERE Id=?', id);
   }
-  listComments(postId: string): Promise<Comment[]> {
-    throw new Error('Method not implemented.');
+
+  async getLikes(postId: string): Promise<number> {
+    let awaitResult = await this.db.get<number>(
+      'Select count(*) FROM Likes WHERE postId=?',
+      postId
+    );
+    let val: number = awaitResult === undefined ? 0 : awaitResult;
```

Comment:
I think this should always return a number right? If there are no likes in the db it'll return 0.